### ISSUE TITLE
can't run on mobile using react-native + expo

### ISSUE THREAD

**Issue by rolandoam** [xmppjs/xmpp.js#536]
I would like to be able to run the pure-js implementation (using websockets, same as in the browser) for an expo app (non-ejected). However, after installing `@xmpp/client`, I'm having tons of requirement issues, where some dependencies of xmpp are trying to require node-specific libraries (crypto comes to mind, randombytes, etc). Is there a way to force xmpp.js to use a specific transport and not try to require those libraries?

**Comment by sonnyp** [comment]
Use https://github.com/webpack/node-libs-browser

**Comment by sonnyp** [comment]
@rolandoam have you been able to test this with your app?

**Comment by soulfly** [comment]
Trying to run the same, but getting

```
18:05:28: error, TypeError: global.btoa is not a function. (In 'global.btoa(string)', 'global.btoa' is undefined)
18:05:29: [Unhandled promise rejection: TypeError: global.btoa is not a function. (In 'global.btoa(string)', 'global.btoa' is undefined)]
* xmpp.js/packages/sasl/lib/b64.js:5:23 in encode
* xmpp.js/packages/sasl/index.js:136:19 in <unknown>
- node_modules/promise/setimmediate/core.js:45:7 in tryCallTwo
- node_modules/promise/setimmediate/core.js:200:23 in doResolve
- node_modules/promise/setimmediate/core.js:66:12 in Promise
* xmpp.js/packages/sasl/index.js:95:23 in authenticate
* xmpp.js/packages/events/lib/EventEmitter.js:20:20 in delegate
- node_modules/promise/setimmediate/core.js:37:14 in tryCallOne
- node_modules/promise/setimmediate/core.js:123:25 in <unknown>
```

Will try your advise with `rn-cli.config.js`

**Comment by soulfly** [comment]
No luck.. looks like the `btoa` is not in that list of ported libs

**Comment by soulfly** [comment]
This code did a fix:

```
  var base64 = require('base-64');
  global.btoa = base64.encode;
  global.atob = base64.decode;
```

**Comment by soulfly** [comment]
I can confirm everything else is working fine on React Native

@sonnyp are you able to make a new release form master? It will be easier to work with it

**Comment by sonnyp** [comment]
@soulfly unfortunately not at the moment, some stuff is still being worked on but I'll try to do ASAP

**Comment by soulfly** [comment]
Why I'm asking about release is because I want to use this lib as a dependency in other lib and then release this other lib, so users will use it in their apps

But I have some difficulties with this  because I do not know how to connect a master branch of xmppjs as a dependency via `package.json` to my lib, looks like it's impossible 

It only works manually this way, but then it's not connected via `package.json` and npm dependencies
```
git clone git@github.com:xmppjs/xmpp.js.git
cd xmpp.js
make
```

**Comment by zek** [comment]
@soulfly having same issue.

**Comment by sonnyp** [comment]
Fixed in 0.5.1 https://github.com/xmppjs/xmpp.js/releases/tag/v0.5.1

**Comment by sonnyp** [comment]
fixed in https://github.com/xmppjs/xmpp.js/releases/tag/v0.9.0
